### PR TITLE
First ComputationNodes, Layers, and Model definition

### PR DESCRIPTION
**Pull request by fedetask**
- Basic ComputationNode derivations that allow fully connected networks with Relu and Softmax activations
- Layer abstract definition and Dense derivation
- First draft of Model

### CODE REVIEW

**Comment by fedetask** [comment]
This pull request should now be ready to be merged as it provides the basic functionalities and abstractions to design neural networks and perform the forward pass.

Labeled as "Ready for review". I'll check things are good and merge it on master

**Comment by fedetask** [comment]
Code seems robust enough and I started developing the backpropagation algorithm. It's time to merge this.
